ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -325,6 +330,14 @@ static VALUE _native_new(VALUE klass) {
 
   reset_stats_not_thread_safe(state);
 
+  long now = monotonic_wall_time_now_ns(DO_NOT_RAISE_ON_FAILURE);
+  if (now == 0) {
+    // NOTE: Throwing an exception on new means leaking memory as noted above so
+    //       lets let things carry on but store a delayed error.
+    delayed_error(state, ERR_CLOCK_FAIL);
```

Comment:
:thinking: Since in this case, the reason to not raise exceptions is to avoid leaking the `state`, we could `free(state)` and then go ahead and raise the exception.

Reply:
Ah yeah wasn't sure if it was just that malloc or there was some other internal thing that would cause a leak around the runtime 😅